scripts/processFactory.ts

```diff
@@ -0,0 +1,525 @@
+﻿/// <reference path="../src/compiler/sys.ts" />
+module ts {
+    interface Map<T> {
+        [name: string]: T;
+    }
+
+    interface SyntaxNode {
+        name?: string;
+        kind?: string;
+        type?: string;
+        types?: string;
+        baseType?: string;
+        children?: SyntaxMember[];
+    }
+
+    interface SyntaxMember {
+        paramName?: string;
+        name?: string;
```

Comment:
I use it for Factory.createNumericLiteral so that it can generate the following:

```TypeScript
        export function createNumericLiteral(value: number | string, location?: TextRange, flags?: NodeFlags): LiteralExpression {
            var node = beginNode<LiteralExpression>(SyntaxKind.NumericLiteral);
            node.text = String(value);
            return finishNode(node, location, flags);
        }
```
Where `paramName` is `"value"` and `name` is `"text"`

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:02 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In scripts/processFactory.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24624892>:

> +    interface Map<T> {

> +        [name: string]: T;

> +    }

> +

> +    interface SyntaxNode {

> +        name?: string;

> +        kind?: string;

> +        type?: string;

> +        types?: string;

> +        baseType?: string;

> +        children?: SyntaxMember[];

> +    }

> +

> +    interface SyntaxMember {

> +        paramName?: string;

> +        name?: string;

What's the difference between paramName and name?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24624892>.